utils/node_local_test.py

```diff
@@ -594,31 +594,30 @@ def start(self):
                              '--undef-value-errors=no']
             self._io_server_dir = tempfile.TemporaryDirectory(prefix='dnt_io_')
 
-            fd = open(os.path.join(self._io_server_dir.name,
-                                   'daos_engine'), 'w')
+            fd = open(join(self._io_server_dir.name, 'daos_engine'), 'w')
             fd.write('#!/bin/sh\n')
             fd.write('export PATH=$REAL_PATH\n')
             fd.write('exec valgrind {} daos_engine "$@"\n'.format(
                 ' '.join(valgrind_args)))
             fd.close()
 
-            os.chmod(os.path.join(self._io_server_dir.name, 'daos_engine'),
+            os.chmod(join(self._io_server_dir.name, 'daos_engine'),
                      stat.S_IXUSR | stat.S_IRUSR)
 
             server_env['REAL_PATH'] = '{}:{}'.format(
-                os.path.join(self.conf['PREFIX'], 'bin'), server_env['PATH'])
+                join(self.conf['PREFIX'], 'bin'), server_env['PATH'])
             server_env['PATH'] = '{}:{}'.format(self._io_server_dir.name,
                                                 server_env['PATH'])
 
-        daos_server = os.path.join(self.conf['PREFIX'], 'bin', 'daos_server')
+        daos_server = join(self.conf['PREFIX'], 'bin', 'daos_server')
 
         self_dir = os.path.dirname(os.path.abspath(__file__))
 
         # Create a server yaml file.  To do this open and copy the
         # nlt_server.yaml file in the current directory, but overwrite
         # the server log file with a temporary file so that multiple
         # server runs do not overwrite each other.
-        scfd = open(os.path.join(self_dir, 'nlt_server.yaml'), 'r')
+        scfd = open(join(self_dir, 'nlt_server.yaml'), 'r')
 
         scyaml = yaml.safe_load(scfd)
```

Comment:
Is this not another case where `with` should be used?
```suggestion
        with open(join(self_dir, 'nlt_server.yaml'), 'r') as scfd:
            scyaml = yaml.safe_load(scfd)
```